Sources/OpenAPIRuntime/URICoder/Common/URICoderConfiguration.swift

```diff
@@ -0,0 +1,72 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftOpenAPIGenerator open source project
+//
+// Copyright (c) 2023 Apple Inc. and the SwiftOpenAPIGenerator project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftOpenAPIGenerator project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import Foundation
+
+/// A bag of configuration values used by the URI parser and serializer.
+struct URICoderConfiguration {
+
+    // TODO: Wrap in a struct, as this will grow.
+    // https://github.com/OAI/OpenAPI-Specification/blob/main/versions/3.0.3.md#style-values
+    enum Style {
+        case simple
+        case form
+    }
+
+    var style: Style
+    var explode: Bool
+    var spaceEscapingCharacter: String
+
+    private init(style: Style, explode: Bool, spaceEscapingCharacter: String) {
+        self.style = style
+        self.explode = explode
+        self.spaceEscapingCharacter = spaceEscapingCharacter
+    }
+
+    static let formExplode: Self = .init(
+        style: .form,
+        explode: true,
+        spaceEscapingCharacter: "%20"
```

Comment:
Minor point, but spaceEscapingCharacter might be better as an enum so we can be sure that we're always using the same value:

```
enum SpaceEscapingCharacter: String {
    case percent = "%20"
    case plus = "+"
}
```

Reply:
Sure, it's not like there'll be a third case any time soon. I'll change it.

Reply:
https://github.com/apple/swift-openapi-runtime/pull/41/commits/a24c805046b2930af0ac919fd44f23f67dc39bb8